### PR TITLE
For #23970, #23969, #23968 - Convert home screen headers to compose

### PR DESCRIPTION
**Pull request by sarah541**
Fixes https://github.com/mozilla-mobile/fenix/issues/23970
Fixes https://github.com/mozilla-mobile/fenix/issues/23969
Fixes https://github.com/mozilla-mobile/fenix/issues/23968

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarksHeaderViewHolder.kt
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarksHeaderViewHolder.kt:26]:
```suggestion
 * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
```

**Review by gabrielluong — APPROVED** [review]

**Comment by gabrielluong** [comment]
Let's wait until Monday to land this. Thanks!

File: app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt:36]: If "show all" is always present, why make the click lambda optional?

File: app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt:47]: There should be a `Spacer` (maybe 16.dp?)  between the header text and show all, just in case some weird header text string or device font size forces them to butt up to each other.

File: app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt
Comment (MozillaNoah) [inline app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt:68]: nit: although we use named parameters almost everywhere, stringResource is probably the one few exceptions when we're not.

**Review by MozillaNoah — DISMISSED** [review]

File: app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt
Comment (sarah541) [reply app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt:46]: Sounds right!

File: app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/compose/list/HomeSectionHeader.kt:51]: This should be width instead of height since it's a Row. Alternatively, I think it might be better to add this as a start padding to the `ClickableText` to increase its tap area.

```suggestion
        Spacer(modifier = Modifier.width(16.dp))
```

**Pull request merged**
